src/hotspot/share/opto/predicates.cpp

```diff
@@ -381,6 +382,86 @@ bool TemplateAssertionPredicateExpressionNode::is_template_assertion_predicate(N
   return node->is_If() && node->in(1)->is_Opaque4();
 }
 
+InitializedAssertionPredicate::InitializedAssertionPredicate(IfNode* template_assertion_predicate, Node* new_init,
+                                                             Node* new_stride, PhaseIdealLoop* phase)
+    : _template_assertion_predicate(template_assertion_predicate),
+      _new_init(new_init),
+      _new_stride(new_stride),
+      _phase(phase) {}
+
+// Create an Initialized Assertion Predicate at the provided control from the _template_assertion_predicate.
+// We clone the Template Assertion Predicate Expression and replace:
+// - Opaque4 with OpaqueInitializedAssertionPredicate
+// - OpaqueLoop*Nodes with _new_init and _new_stride, respectively.
+//
+//          init                 stride
+//            |                    |
+//   OpaqueLoopInitNode  OpaqueLoopStrideNode                      _new_init    _new_stride
+//                  \     /                                               \     /
+//                    ...                                                   ...
+//                     |                                                     |
+//                    Bool                                                new Bool
+//                     |                                                     |
+//                  Opaque4                ======>              control   OpaqueInitializedAssertionPredicate
+//                     |                                              \  /
+//                    If                                             new If
+//                  /    \                                           /    \
+//            success     fail path                         new success   new Halt
+//              proj    (Halt or UCT)                           proj
+//
+IfTrueNode* InitializedAssertionPredicate::create(Node* control) {
+  IdealLoopTree* loop = _phase->get_loop(control);
+  OpaqueInitializedAssertionPredicateNode* assertion_expression = create_assertion_expression(control);
+  IfNode* if_node = create_if_node(control, assertion_expression, loop);
+  create_fail_path(if_node, loop);
+  return create_success_path(if_node, loop);
+}
+
+// Create a new Assertion Expression to be used as bool input for the Initialized Assertion Predicate IfNode.
+OpaqueInitializedAssertionPredicateNode* InitializedAssertionPredicate::create_assertion_expression(Node* control) {
+  Opaque4Node* template_opaque = _template_assertion_predicate->in(1)->as_Opaque4();
+  TemplateAssertionPredicateExpression assertion_expression(template_opaque);
+  Opaque4Node* tmp_opaque =
+      assertion_expression.clone_and_replace_init_and_stride(_new_init, _new_stride, control, _phase);
+  OpaqueInitializedAssertionPredicateNode* assertion_predicate_expression
+      = new OpaqueInitializedAssertionPredicateNode(tmp_opaque->in(1)->as_Bool(), _phase->C);
```

Comment:
Decide if you want the `=` on the first or last line and do it consistently ;)

Reply:
Long names are so nice, but the code does get a big difficult to format :sweat_smile: